File: runtime/java/titan-ast-runtime/src/titan/ast/runtime/TokenDfaState.java
package titan.ast.runtime;

import java.util.HashMap;
import java.util.Map;

/**
 * token的 确定有限状态自动机的 状态.
 *
 * @author tian wei jun
 */
class TokenDfaState {
  int type = FaStateType.NONE.getValue();
  int weight = 0;
  // token语法名字
  Grammar terminal = null;
  // 转移
  Map<Integer, TokenDfaState> edges = new HashMap<>();
}

File: runtime/cpp/titan-ast-runtime-lib/TokenDfaState.cpp
//
// Created by tian wei jun on 2022/11/30 0030.
//

#include "TokenDfaState.h"

TokenDfaState::TokenDfaState()
    : type((int) FaStateEnumType::NONE), weight(0), terminal(nullptr),
      edges(std::unordered_map<byte, TokenDfaState *>()) {}
// terminal delete by PersistentData.grammars
// TokenDfaState in edges delete by TokenDfa
TokenDfaState::~TokenDfaState() = default;